### PR TITLE
Date pickers

### PR DESCRIPTION
**Pull request by natjoe4**
# Summary
Added measurement period date pickers and autofill functionality on measure upload

## New Behavior
- Interactive date pickers now available on home screen
- On measure bundle upload, measurementPeriod date pickers will autofill to the `effectivePeriod` of the measure resource. If effective period is not defined, the period will default to `2022-01-01 - 2022-12-31`, If only the start or end of the effectivePeriod is defined, the datePickers will be populated with a 1 year period based on the value of the defined point.

## Code Changes
- Added DateSelectors component
- Added populateMeasurePeriod function which gets called on measureBundle upload
- Added testing
- Added luxon, dayjs, and @mantine/dates as dependencies

# Testing Guidance
- Run `npm test`
- Start up the app with `npm run dev`
- Upload the EXM130 bundle and ensure the date pickers auto fill to `January, 1 2018` and `December 31, 2018`
- Remove the `effectivePeriod` attribute on the EXM130 measure, then re-upload. Ensure the datePickers update to `January 1, 2022` and `December 31, 2022`
- Add back just the end date, and ensure the date pickers populate to `December 31, 2017` `December 31, 2018`
- Add back just the start date, and ensure the date pickers populate to `January 1, 2018` `January 1, 2019`
- Select dates for yourself and ensure they overwrite the auto-poulated dates

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,13 +16,16 @@
...
   "dependencies": {
     "@mantine/core": "^4.2.4",
+    "@mantine/dates": "^4.2.6",
     "@mantine/dropzone": "^4.2.5",
     "@mantine/hooks": "^4.2.4",
     "@mantine/next": "^4.2.4",
     "@mantine/notifications": "^4.2.5",
     "@mantine/prism": "^4.2.4",
     "@tabler/icons": "^1.68.0",
+    "dayjs": "^1.11.2",
```
Comment (sarahmcdougall) [inline package.json:26]: Where does `dayjs` get used/what is it used for?

File: components/MeasureUpload.tsx [hunk 2]
```diff
@@ -81,3 +96,40 @@ function DropzoneChildren() {
...
+    if (endString) {
+      measurementPeriod.end = DateTime.fromISO(endString).toJSDate();
+      // If start is defined and end isn't, make the period last one year from the start
+    } else {
+      const newEnd = new Date(startDate);
+      newEnd.setFullYear(startDate.getFullYear() + DEFAULT_MEASUREMENT_PERIOD_DURATION);
+      measurementPeriod.end = newEnd;
+    }
+
+    // If end is defined and start isn't, make the period start one year the end
```
Comment (sarahmcdougall) [inline components/MeasureUpload.tsx:126]:
```suggestion
    // If end is defined and start isn't, make the period start one year from the end
```

**Review by sarahmcdougall — CHANGES_REQUESTED** [review]
Works well and unit tests pass :) Code is commented really nicely so that it's clear what's going on when we have the period start selected v. period end selected v. both selected. Just a few comments regarding the date dependencies

File: package.json (hunk 1 above)
Comment (natjoe4) [reply package.json:26]: dayjs is a required peer dependency for @mantine/date :)

**Pull request merged**